common-healing.lic

```diff
@@ -20,239 +75,949 @@ $wound_map = {
+  # yourself with medicinal potions and salves, or healing spells.
   def check_health
-    all_parasites_re = Regexp.union($all_parasites)
+    parasites_regex = Regexp.union($DRCH_PARASITES_REGEX_LIST)
     wounds_line = nil
     parasites_line = nil
+    lodged_line = nil
     diseased = false
     poisoned = false
 
     DRC.bput('health', 'You have', 'You feel somewhat tired and seem to be having trouble breathing')
     pause 0.5
-    data = reget(50).reverse
-    data.each do |line|
+    health_lines = reget(50).map(&:strip).reverse
+
+    health_lines.each do |line|
       case line
-      when /^You have (?!no significant injuries)(?!.* lodged into your)(?!.* infection)(?!.* poison(?:ed)?)(?!.* #{all_parasites_re})/
-        wounds_line = line
       when /^Your body feels\b.*(?:strength|battered|beat up|bad shape|death's door|dead)/
+        # While iterating the game output in reverse,
+        # this line represents the start of `HEALTH` command
+        # so we can now break out of the loop and avoid
+        # parsing text prior to the latest HEALTH command.
+        # We don't need to parse this line, just use implicit `health` numeric variable.
         break
+      when /Your spirit feels\b.*(?:mighty and unconquerable|full of life|strong|wavering slightly|shaky|weak|cold|empty|lifeless|dead)/
+        # This line is your spirit health.
+        # We don't need to parse this line, just use implicit `spirit` numeric variable.
+        next
+      when /^You have (?!no significant injuries)(?!.* lodged .* into your)(?!.* infection)(?!.* poison(?:ed)?)(?!.* #{parasites_regex})/
+        wounds_line = line
+      when /^You have .* lodged .* into your/
+        lodged_line = line
+      when /^You have a .* on your/, parasites_regex
+        parasites_line = line
       when /^You have a dormant infection/, /^Your wounds are infected/, /^Your body is covered in open oozing sores/
         diseased = true
       when /^You have .* poison(?:ed)?/, /^You feel somewhat tired and seem to be having trouble breathing/
         poisoned = true
-      when /^You have a .* on your/, all_parasites_re
-        parasites_line = line
       end
     end
 
+    bleeders = parse_bleeders(health_lines)
+    wounds = parse_wounds(wounds_line)
+    parasites = parse_parasites(parasites_line)
+    lodged_items = parse_lodged_items(lodged_line)
+
+    return {
+      'wounds' => wounds,
+      'bleeders' => bleeders,
+      'parasites' => parasites,
+      'lodged' => lodged_items,
+      'poisoned' => poisoned,
+      'diseased' => diseased
+    }
+  end
+
+  # Uses PERCEIVE HEALTH SELF command to check for wounds and scars.
+  # Returns a map of any wounds where the keys
+  # are the severity and the values are the list of wounds.
+  def perceive_health
+    return unless DRStats.empath?
+
+    wounds = Hash.new { |h, k| h[k] = [] }
+
+    DRC.bput('perceive health self', 'Roundtime')
```

Comment:
Per [Kiriawen](https://discord.com/channels/745675889622384681/745696628714897508/788819167351406592), handle the following message that shocked empaths receive when they try to perceive their own health:

```
>perceive health self
You close your eyes, centering your thoughts on your own life essence, but feel only an aching emptiness.
```

Reply:
Is that a change you'll be making in this PR? If so, could you make this a draft so I don't accidentally merge it?

Reply:
resolved per https://github.com/rpherbig/dr-scripts/pull/4659/commits/c7beadb5425984219c4b106fd8ab0c7401d008ab